src/mam4xx/mam4_amicphys.hpp

```diff
@@ -0,0 +1,2371 @@
+    //  Partition mass and number before cloud chemistry
+    //  NOTE that in this case jclea is 1 and jcldy is 2
+    //  Following 2 calls set qqcwsub2(:,1)=0 and qqcwsub2(:,2) to a computed
+    //  value
+    set_subarea_qnumb_for_cldbrn_aerosols(jclea, jcldy, fcldy,
+                                          qqcwgcm2,  // in
+                                          qqcwsub2); // inout
+
+    set_subarea_qmass_for_cldbrn_aerosols(jclea, jcldy, fcldy,
+                                          qqcwgcm2,  // in
+                                          qqcwsub2); // inout
+    //  Partition mass and number before cloud chemistry
+    // Following 2 calls set qqcwsub3(:,1)=0 and qqcwsub3(:,2) to a computed
+    // value
+    set_subarea_qnumb_for_cldbrn_aerosols(jclea, jcldy, fcldy,
+                                          qqcwgcm3,  // in
+                                          qqcwsub3); // inout
+    set_subarea_qmass_for_cldbrn_aerosols(jclea, jcldy, fcldy,
+                                          qqcwgcm3,  // in
+                                          qqcwsub3); // inout
+
+    //----------------------------------------------------------------------------------------
+    // Interstitial aerosols. (They can exist in both cloudy and clear
+    // subareas, and hence need to be partitioned.)
+    //----------------------------------------------------------------------------------------
+    // Partition mass and number before cloud chemistry
+    // Following 2 calls set qsub2(:,1) = 0 and qsub2(:,2) to a computed value
+    set_subarea_qnumb_for_intrst_aerosols(jclea, jcldy, fclea, fcldy, // in
+                                          qgcm2, qqcwgcm2, qgcm2,     // in
+                                          qsub2);                     // inout
+
+    set_subarea_qmass_for_intrst_aerosols(jclea, jcldy, fclea, fcldy, // in
+                                          qgcm2, qqcwgcm2, qgcm2,     // in
+                                          qsub2);                     // inout
+
+    // Partition mass and number before cloud chemistry
+    // Following 2 calls set qsub3(:,1) = 0 and qsub3(:,2) to a computed value
+    set_subarea_qnumb_for_intrst_aerosols(jclea, jcldy, fclea, fcldy, // in
+                                          qgcm2, qqcwgcm2, qgcm3,     // in
+                                          qsub3);                     // inout
+
+    set_subarea_qmass_for_intrst_aerosols(jclea, jcldy, fclea, fcldy, // in
+                                          qgcm2, qqcwgcm2, qgcm3,     // in
+                                          qsub3);                     // inout
+
+  } // different categories
+} // set_subarea_gases_and_aerosols
+
+//--------------------------------------------------------------------------------
+//--------------------------------------------------------------------------------
+
+KOKKOS_INLINE_FUNCTION
+void mam_newnuc_1subarea(
```

Comment:
This is ported, I believe, as one of 2 versions of `nucleation::compute_tendencies()`--see other comment at where this gets called: $\approx 1595$

Reply:
This `mam_newnuc_1subarea` is the direct port of the Fortran code in e3sm-refactor repo.